multi-node/aws/pkg/config/config_test.go

```diff
@@ -417,3 +417,185 @@ subnets:
 	}
 
 }
+
+func TestControllerVolumeType(t *testing.T) {
+
+	validConfigs := []struct {
+		conf       string
+		volumeType string
+		iops       int
+	}{
+		{
+			conf:       ``,
+			volumeType: "gp2",
+			iops:       0,
+		},
+		{
+			conf: `
+controllerRootVolumeType: gp2
+`,
+			volumeType: "gp2",
+			iops:       0,
+		},
+		{
+			conf: `
+controllerRootVolumeType: standard
+`,
+			volumeType: "standard",
+			iops:       0,
+		},
+		{
+			conf: `
+controllerRootVolumeType: io1
+controllerRootVolumeIOPS: 100
+`,
+			volumeType: "io1",
+			iops:       100,
+		},
+		{
+			conf: `
+controllerRootVolumeType: io1
+controllerRootVolumeIOPS: 2000
+`,
+			volumeType: "io1",
+			iops:       2000,
+		},
+	}
+
+	invalidConfigs := []string{
+		`
+controllerRootVolumeType: default
+`,
+		`
+controllerRootVolumeType: standard
+controllerRootVolumeIOPS: 100
+`,
+		`
+controllerRootVolumeType: gp2
+controllerRootVolumeIOPS: 2000
+`,
+		`
+controllerRootVolumeType: io1
+controllerRootVolumeIOPS: 99
+`,
+		`
+controllerRootVolumeType: io1
+controllerRootVolumeIOPS: 2001
+`,
+	}
+
+	for _, conf := range validConfigs {
+		confBody := singleAzConfigYaml + conf.conf
+		c, err := ClusterFromBytes([]byte(confBody))
+		if err != nil {
+			t.Errorf("failed to parse config %s: %v", confBody, err)
+			continue
+		}
+		if c.ControllerRootVolumeType != conf.volumeType {
+			t.Errorf(
+				"parsed release channel %s does not match config: %s",
+				c.ReleaseChannel,
+				confBody,
+			)
+		}
+	}
+
+	for _, conf := range invalidConfigs {
+		confBody := singleAzConfigYaml + conf
+		_, err := ClusterFromBytes([]byte(confBody))
+		if err == nil {
+			t.Errorf("expected error parsing invalid config: %s", confBody)
+		}
+	}
+}
+
+func TestWorkerVolumeType(t *testing.T) {
+
+	validConfigs := []struct {
+		conf       string
+		volumeType string
+		iops       int
+	}{
+		{
+			conf:       ``,
+			volumeType: "gp2",
+			iops:       0,
+		},
+		{
+			conf: `
+workerRootVolumeType: gp2
+`,
+			volumeType: "gp2",
+			iops:       0,
+		},
+		{
+			conf: `
+workerRootVolumeType: standard
+`,
+			volumeType: "standard",
+			iops:       0,
+		},
+		{
+			conf: `
+workerRootVolumeType: io1
+workerRootVolumeIOPS: 100
+`,
+			volumeType: "io1",
+			iops:       100,
+		},
+		{
+			conf: `
+workerRootVolumeType: io1
+workerRootVolumeIOPS: 2000
+`,
+			volumeType: "io1",
+			iops:       2000,
+		},
+	}
+
+	invalidConfigs := []string{
```

Comment:
each invalid config string should contain within it a short comment denoting why that config is invalid.